src/internal/agent/hooks/flux.go

```diff
@@ -26,35 +28,71 @@ type GenericGitRepo struct {
 func NewGitRepositoryMutationHook() operations.Hook {
 	message.Debug("hooks.NewGitRepositoryMutationHook()")
 	return operations.Hook{
-		Create: mutateGitRepository,
-		Update: mutateGitRepository,
+		Create: mutateGitRepoCreate,
+		Update: mutateGitRepoUpdate,
+	}
+}
+
+func mutateGitRepoCreate(r *v1.AdmissionRequest) (*operations.Result, error) {
+	var patches []operations.PatchOperation
+
+	zarfState, err := getStateFromAgentPod(zarfStatePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load zarf state from file: %w", err)
 	}
+
+	// Form the gitServerURL from the state
+	gitServerURL := zarfState.GitServer.Address
+	message.Debugf("Using the gitServerURL of (%s) to mutate the flux repository", gitServerURL)
+
+	// parse to simple struct to read the git url
+	gitRepo := &GenericGitRepo{}
+	if err := json.Unmarshal(r.Object.Raw, &gitRepo); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal manifest: %w", err)
+	}
+
+	replacedURL := git.MutateGitUrlsInText(gitServerURL, gitRepo.Spec.URL, zarfState.GitServer.PushUsername)
+	message.Debugf("original git URL of (%s) got mutated to (%s)", gitRepo.Spec.URL, replacedURL)
+
+	// Patch updates of the repo spec
+	patches = populatePatchOperations(replacedURL, gitRepo.Spec.SecretRef.Name)
+
+	return &operations.Result{
+		Allowed:  true,
+		PatchOps: patches,
+	}, nil
 }
 
-func mutateGitRepository(r *v1.AdmissionRequest) (*operations.Result, error) {
+func mutateGitRepoUpdate(r *v1.AdmissionRequest) (*operations.Result, error) {
 	var patches []operations.PatchOperation
 
 	// parse to simple struct to read the git url
 	gitRepo := &GenericGitRepo{}
 	if err := json.Unmarshal(r.Object.Raw, &gitRepo); err != nil {
-		return nil, fmt.Errorf("failed to unmarshal manifest: %v", err)
+		return nil, fmt.Errorf("failed to unmarshal manifest: %w", err)
 	}
 
-	message.Info(gitRepo.Spec.URL)
+	// Patch updates of the repo spec
+	patches = populatePatchOperations(gitRepo.Spec.URL, gitRepo.Spec.SecretRef.Name)
 
-	replacedURL := git.MutateGitUrlsInText("http://zarf-gitea-http.zarf.svc.cluster.local:3000", gitRepo.Spec.URL)
```

Comment:
What if someone updates the upstream URL for something they deploy?  i.e. instead of getting big bang from https://github.com/DoD-Platform-One/big-bang.git they later change that to https://repo1.dso.mil/platform-one/big-bang/bigbang.git?  (Theoretically, I haven't fully tested this) The name of the GitRepository resource wouldn't be different and so would trigger an update but that URL would never be swapped leading to people (potentially) scratching their heads.  I think it would be good to have a check like we used to have, but instead of looking for the git user, we look for the airgapped git url since we can be (reasonably) certain that that is unique to the air gap.

Reply:
Just to make sure I'm following, are you suggesting we check if the hostname is matching the hostname we have saved in the Zarf state?

Reply:
yes

Reply:
Whoops I did this yesterday but forgot to push apparently... This should be checking the hostname like you recommended now.

Reply:
That's a very good point, this would be much better.